packages/suite/src/views/wallet/transactions/components/TransactionList/index.tsx

```diff
@@ -81,6 +91,127 @@ const TransactionList = ({
         slicedTransactions,
     ]);
 
+    const { translationString } = useTranslation();
+    const { addNotification } = useActions({
+        addNotification: addToast,
+    });
+    const [isExportRunning, setIsExportRunning] = useState(false);
+    const { symbol, descriptor } = account;
+    const runExport = useCallback(
+        async type => {
+            // Don't run again if already running
+            if (isExportRunning) {
+                return;
+            }
+
+            // Set as in running
+            setIsExportRunning(true);
+
+            // Fetch the first page of transactions (this will tell us how many pages exist)
+            const result = await TrezorConnect.getAccountInfo({
+                descriptor,
+                coin: symbol,
+                details: 'txs',
+            });
+
+            // If that failed, notify the user that it couldn't run the export
+            if (!result.success) {
+                addNotification({ type: 'error', error: translationString('TR_EXPORT_FAIL') });
+                setIsExportRunning(false);
+                return;
+            }
+
+            const { history, page } = result.payload;
+            const transactions = history.transactions ?? [];
+
+            // If there are more than one page, fetch all pages
+            if (page && page.total > 1) {
+                const pages = range(2, page.total);
+                const promises = pages.map(p =>
+                    TrezorConnect.getAccountInfo({
+                        descriptor,
+                        coin: symbol,
+                        details: 'txs',
+                        page: p,
+                    }),
+                );
+                const results = await Promise.all(promises);
+                results.forEach(r => {
+                    if (!r.success) {
+                        addNotification({
+                            type: 'error',
+                            error: translationString('TR_EXPORT_FAIL'),
+                        });
+                        setIsExportRunning(false);
+                        return;
+                    }
+
+                    if (r.payload.history.transactions) {
+                        transactions.push(...r.payload.history.transactions);
```

Comment:
it would be nice to store dowloaded data in indexed db, for later use
in fact, fetching pages should check if the page that you want to download is already present in indexed db

Reply:
Kinda disagree on this one. If you have a lot of transactions, it will use up quite some memory for something you will be rarely be needing again. On the other hand, I think we decide to cache transactions, it would be better to start with the transactions page and then use the already fetched transactions for the export.

Reply:
there is a special logic for that, `transactions in indexed db` != `transactions in reducer` to avoid leaks and not to keep them all in javascript memory.
basically reducer keeps only 1 page, and fetch next requested page either from indexeddb (if exists) or from the TrezorConnect

Reply:
but this of course can be addressed in another PR

Reply:
Interesting, I'd appreciate if you show me around that part of the project when you have time! Would be useful for the search feature as well.